app/src/main/java/com/android/unio/model/search/SearchViewModel.kt

```diff
@@ -60,32 +65,44 @@ class SearchViewModel(private val repository: SearchRepository) : ViewModel() {
       val results = repository.searchAssociations(query)
       _associations.value = results
       status.value = Status.SUCCESS
-      Log.d("SearchViewModel", "searchAssociations: $results")
     }
   }
 
   /**
    * Debounces the search query to avoid making too many requests in a short period of time.
    *
    * @param query The query to search for.
+   * @param searchType The type of search to perform.
    */
-  fun debouncedSearch(query: String) {
+  fun debouncedSearch(query: String, searchType: SearchType) {
     searchJob?.cancel()
-    searchJob =
-        viewModelScope.launch {
-          delay(500)
-          if (query.isNotEmpty()) {
-            searchAssociations(query)
-          } else {
-            clearAssociations()
+    if (query.isEmpty()) {
+      when (searchType) {
+        SearchType.EVENT -> clearEvents()
+        SearchType.ASSOCIATION -> clearAssociations()
+      }
+    } else {
+      searchJob =
+          viewModelScope.launch {
+            delay(500)
+            when (searchType) {
+              SearchType.EVENT -> searchEvents(query)
+              SearchType.ASSOCIATION -> searchAssociations(query)
+            }
           }
-        }
+    }
   }
 
   /** Clears the list of associations and sets the search status to [Status.IDLE]. */
-  fun clearAssociations() {
+  private fun clearAssociations() {
     _associations.value = emptyList()
     status.value = Status.IDLE
+    Log.d("SearchViewModel", status.value.toString())
```

Comment:
I guess this logging line was for debugging ? If it is the case, this should be removed